src/components/generic/ModalPopup.tsx

```diff
@@ -0,0 +1,116 @@
+import {
+    KeyboardEvent,
+    MouseEvent,
+    ReactNode,
+    TouchEvent,
+    useRef,
+} from 'react';
+import classes from './ModalPopup.module.scss';
+
+import { X } from '@phosphor-icons/react';
+import { createPortal } from 'react-dom';
+import classNames from 'classnames/bind';
+import Button from '../input/button/Button';
+
+const cx = classNames.bind(classes);
+
+export type ButtonType = 'primary' | 'secondary' | 'close';
+export type onButtonClick = (button: ButtonType) => void;
+
+interface ModalPopupProps {
+    title: string;
+    primaryButtonText: string;
+    secondaryButtonText: string;
+    disabledPrimaryButton?: boolean;
+    dangerousSecondaryButton?: boolean;
+    onButtonClick: onButtonClick;
+    children: ReactNode;
+}
+
+const CLOSE_ICON_PROPS = {
+    size: 30,
+};
+
+const ModalPopup = ({
+    title,
+    primaryButtonText,
+    secondaryButtonText,
+    disabledPrimaryButton,
+    dangerousSecondaryButton,
+    onButtonClick,
+    children,
+}: ModalPopupProps) => {
+    const ref = useRef<HTMLDivElement>(null);
+
+    const cantUsePrimaryButton =
+        typeof disabledPrimaryButton === 'boolean' &&
+        disabledPrimaryButton === true;
```

Comment:
This is redundant, since it can either be boolean or undefined, and undefined is by definition falsey.